ext/curl/Kconfig

```diff
@@ -10,7 +10,7 @@ config NRF_CURL_PROMPTLESS
 
 config NRF_CURL
 	bool "Curl" if !NRF_CURL_PROMPTLESS
-	depends on POSIX_API && !PTHREAD_IPC && NEWLIB_LIBC
+	depends on POSIX_API && !POSIX_THREADS && NEWLIB_LIBC
```

Comment:
Remove the dependency to `PTHREAD_IPC` rather than `POSIX_THREADS` and Curl seems to work with `modem_shell` sample. So this line should be:
`depends on POSIX_API && NEWLIB_LIBC`

Reply:
Added commit to remove these. It still lacks justification as to why it was there and why it can now be removed.

Reply:
@jhirsi is the author of the work porting curl/iperf3 to NCS. The comment from him last week was that `!PTHREAD_IPC` was there likely because compilation failed without this dependency. So it may just be that while `PTHREAD_IPC` got deprecated, some of it's dependencies had been cleaned up and POSIX_THREADS doesn't have all the same dependencies so everything works nicely now. They might have worked for some time.
I don't have a better answer than that without going through all related changes in upstream and trying when things started to succeed without this dependency.